sirius_control/sirius_control/sirius_status.py

```diff
@@ -0,0 +1,78 @@
+import rclpy
+from rclpy.node import Node
+from industrial_msgs.msg import RobotStatus, TriState, RobotMode
+from std_msgs.msg import String
+
+
+class SiriusStatus(Node):
+
+    def __init__(self):
+        super().__init__('sirius_status')
+        self.send_status = self.declare_parameter('send_status', '/sirius_status').value
+        self.default_receive_topics = [
+            '/joy_multiplexer/selected_output',
+            '/relaxing_middleware/state'
+        ]
+        self.receive_topics = self.declare_parameter('receive_topics', self.default_receive_topics).value
+        self.initialize_robot_status()
+        self.create_subscriptions()
+        self.publisher = self.create_publisher(RobotStatus, self.send_status, 10)
+        
+    def initialize_robot_status(self):
+        self.robot_status = RobotStatus()
+        self.robot_status.e_stopped.val = TriState.FALSE
+        self.robot_status.drives_powered.val = TriState.TRUE
+        self.robot_status.motion_possible.val= TriState.TRUE
+        self.robot_status.in_motion.val = TriState.FALSE
+        self.robot_status.in_error.val = TriState.FALSE
+        self.robot_status.mode.val = RobotMode.UNKNOWN
+        self.robot_status.error_codes = []
+
+
+    def topic_callback(self, msg):
+        self.get_logger().info('Received message: "%s"' % msg.data)
+        if msg.data == "Locked":
+            self.robot_status.in_motion.val = TriState.FALSE
+            self.robot_status.drives_powered.val = TriState.TRUE
+
+        elif msg.data == "__none":
+            self.robot_status.mode.val = RobotMode.UNKNOWN
+
+        elif msg.data == "joy_diff_drive":
+            self.robot_status.mode.val = RobotMode.MANUAL
+
+        elif msg.data == "autonomic":
+            self.robot_status.mode.val = RobotMode.AUTO
```

Comment:
Using a single callback for two (semantically) different message types is an interesting approach, but equally confusing. Consider making one callback for joy_multiplexer messages and another one for robot status message.

Reply:
Resloved in 5355693